loafer/routes.py

```diff
@@ -51,18 +51,14 @@ def __str__(self):
             type(self).__name__, self.name, self.provider, self.handler)
 
     def apply_message_translator(self, message):
-        processed_message = {'content': message,
-                             'metadata': {}}
         if not self.message_translator:
-            return processed_message
+            return {'content': message, 'metadata': {}}
 
-        translated = self.message_translator.translate(processed_message['content'])
-        processed_message['metadata'].update(translated.get('metadata', {}))
-        processed_message['content'] = translated['content']
-        if not processed_message['content']:
+        translated = self.message_translator.translate(message)
+        if not translated['content']:
             raise ValueError('{} failed to translate message={}'.format(self.message_translator, message))
 
-        return processed_message
+        return translated
```

Comment:
This might override metadata values if we apply the translator several times. I'll check and then we can improve this on the side.